### ISSUE TITLE
Wrong relation in tlinks between events and t0 (dct)

### ISSUE THREAD

**Issue by assaft** [assaft/caevo#15]
Several examples here:, notice the links in bold. In general I assume that the dct is interpreted as the time of utterance which is at preset tense. Every event that took place in the past is supposed to precede it.

Test t08: _They didn't expect the storm, although a report had been sent 2 hours before._
Caevo returns:
ei1,expect
ei2,sent
t0,2002-07-10
t1,PT2H
l1,ei2,t1,BEFORE
l2,ei1,t1,AFTER
l3,ei1,ei2,AFTER
l4,ei2,t0,BEFORE
**l5,ei1,t0,VAGUE**
l6,t1,t0,VAGUE

Test t16:   _John caused the fire._
Caevo returns:
ei1,caused
t0,2002-07-10
**l1,ei1,t0,VAGUE**

See also Issue #14.

**Comment by tc64** [comment]
Assaft, I think I know what is going on here. If you're using either rules or a trained model based on the TimeBank-Dense corpus, CAEVO may be treating the DCT as the entire day on which the document was published. If that's not what's going on, then perhaps the rest of this post is not relevant.

If the DCT was treated as a day during training, there are cases in TimeBank-Dense (the corpus the included models were trained on) where for the same e,t pair, one annotator said BEFORE where another said DURING, because the latter annotator thought it was likely that the event occurred on the same day as the DCT. Due to a disagreement, the ground truth corpus says VAGUE for such pairs. I don't know how many cases there were like this, but, you can check in the training data since there should be files for different annotators for a given document.

Ideally, we would have annotated for both the DCT as day and DCT as moment cases separately, and the former would have lots of VAGUE relations with events while the latter would be more straightforward. If you want to train a model that treats DCT as a moment, you can check the Temporal-2 and Tempeval-3 datasets. If you have access to LDC corpora, you can check various Automated Content Extraction datasets - the event attribute "tense" in those datasets actually means the temporal relation with the DCT (again, conceptualized as a moment as opposed to the entire day).

**Comment by ymishory** [comment]
I'm adding a new sieve that will link the DCT to as-yet-unlinked events, based on their tense (assuming the DCT is the present). It will run after the major rule-based sieves.

**Comment by assaft** [comment]
tc64 - Thanks very much for the explanation. And yes, things make sense now, since indeed we treat the DCT as a full day. 
Yuval - As we discussed, I don't think we can count on the tense of a verb to indicate whether it took place before or after the day indicated by the DCT. Past tense can simply mean 'before now', without any implication that should mark it as 'before today'. Same goes for future tense. 

My conclusions about the DCT are:
1. The DCT can help to resolve co-referring time expressions. For example, in _We paid the bills yesterday_ we can find what _yesterday_ means based on what the DCT is, and the same goes in more complex cases like _next year_, etc.
2. The DCT can help to set tlink relations based on absolute time references. For example _John gave Mary a letter that Bill wrote in 2005_. Assuming the DCT is somewhere in 2006, we can conclude that the 'gave' event is after the 'wrote' event.
3. It is unsafe to set relations between events and t0 based on tense/aspect/class only. VAGUE is the only safe option.

I will update the test suit accordingly.

**Comment by ymishory** [comment]
I agree with the analysis. However, I think we should keep the code that
uses the event attributes to determine the relations type to the DCT. We
should use ablation, once we have the new gold standard ready, to determine
if it helps or harms the overall accuracy of CAEVO 1.2.
